### DISCUSSION TITLE
Can't get all Users have Roles

### DISCUSSION THREAD

**Discussion by newgoldenman** [spatie/laravel-permission#1842] [Q&A]
Hi

I've installed and i am using this very nice package.

everything is ok. 
i use it on laravel 8.

**I Can Do Everything such as create or delete Roles or Permissions. These are OK.**

BUT
I have a big problem:

**i cant get ALL USERS have a specific Role for example: 'writer':**

`$users = User::role('writer')->get(); // Returns only users with the role 'writer'`
This code wont work for me.

The error is:
```
BadMethodCallException
Call to undefined method Illuminate\Foundation\Auth\User::role()
```

and on phpStorm this error is:
`method 'role' not found in Illuminate\Foundation\Auth\User`

Please help me.
Thanks a lot.

**Comment by erikn69** [comment]
Are you using the trait like documentation says?

**Reply by newgoldenman** [reply]
yes i do. i said, everything is ok, but just 2 things are not working:
List of Users have a specific Role
List of Users have a specific permission

I solved my problem. i wrote 2 helper function to have this 2 things.

anyway thanks your nice extension.

**Reply by erikn69** [reply]
weird, there's a lot of tests on that working
https://github.com/spatie/laravel-permission/blob/c54a494baa6c6f1ac63697ecfeffd2bb2260cf09/tests/HasRolesTest.php#L315-L418
